File: competitive-programming/solutions/leetcode/349-intersection-of-two-arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        my_set = set([])
        for i in nums1:
            for j in nums2:
                if i==j:
                    my_set.add(i)
        return my_set

        

File: competitive-programming/solutions/leetcode/349-intersection-of-two-arrays.cpp
class Solution
{
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
    {
        set<int> s;
        vector<int> v;

        for (auto e : nums1)
            s.insert(e);
        for (auto e : nums2)
        {
            auto f = s.find(e);
            if (f != s.end())
            {
                s.erase(f);
                v.push_back(e);
            }
        }
        return v;
    }
};
